### PR TITLE
Switch to gcc for future 2.7.x builds

### PR DESCRIPTION
**Pull request by borrrden**
Without these changes the build will fail on CentOS 6 because the `clock_gettime` function used by CivetWeb will not be found.

### CODE REVIEW

**Comment by borrrden** [comment]
Found a bit more after attempting to build on multiple machines.  My guess of the difference is that the PR validation machine has two versions of GCC installed and so is possibly sharing the linker between them as the system installed version.  Either way, I was able to fix it on that machine and the official docker build image is happy too.

File: LiteCore/tests/CMakeLists.txt [hunk 1]
```diff
@@ -102,3 +100,5 @@ target_link_libraries(
     Support
     ${LITECORE_CRYPTO_LIB}
 )
+
```
Comment (bmeike) [inline LiteCore/tests/CMakeLists.txt:103]: Why is this necessary?  ... or, perhaps, just desirable?

File: C/tests/cmake/platform_linux.cmake [hunk 2]
```diff
@@ -33,5 +33,6 @@ function(setup_build)
         ${LIBCXXABI_LIB}
         ${ZLIB_LIB}
         dl
+        rt
```
Comment (bmeike) [inline C/tests/cmake/platform_linux.cmake:36]: What is "rt".

**Review by bmeike — COMMENTED** [review]
There only think I understand looks right.  Can you say what the other two changes are about?

**Review by bmeike — APPROVED** [review]

File: C/tests/cmake/platform_linux.cmake (hunk 2 above)
Comment (borrrden) [reply C/tests/cmake/platform_linux.cmake:36]: Yes, it is the library for [POSIX.1b realtime extensions](https://www.unix.com/man-page/sunos/3lib/librt/)

File: LiteCore/tests/CMakeLists.txt (hunk 1 above)
Comment (borrrden) [reply LiteCore/tests/CMakeLists.txt:103]: It is so that things like ICU and such show up _after_ everything else to satisfy GCC's linker which only supplies symbols to satisfy previously undefined ones in the linker input.  Without this things like deflate, and ICU functions show up as "undefined reference" and the link fails.  This was addressed in a more robust way in 2.8+.

Comment (bmeike) [reply LiteCore/tests/CMakeLists.txt:103]: Got it.  ... and the 2.8+ way is to big a change to put here?

Comment (borrrden) [reply LiteCore/tests/CMakeLists.txt:103]: It could be done but it didn't seem worth it because the 2.8 change was basically a CMake project overhaul.  This change accomplishes what is needed while keeping changes to a minimum.

Comment (bmeike) [reply LiteCore/tests/CMakeLists.txt:103]: Yep.  Got it.  Approved.
